openapi.yaml

```diff
@@ -0,0 +1,92 @@
+openapi: 3.1.0
+info:
+  version: 0.1.0
+  title: AsyncAPI Server API
+  description: Server API providing official AsyncAPI tools
+  contact:
+    name: AsyncAPI Initiative
+    email: info@asyncapi.io
+    url: https://asyncapi.com/
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+servers:
+  - url: http://api.asyncapi.io
+
+paths:
+  /generate:
+    summary: Group of paths that supports the AsyncAPI Generator - https://www.asyncapi.com/tools/generator.
+    post:
+      summary: Generate the given AsyncAPI template.
+      operationId: generateTemplate
+      tags:
+        - generator
+      externalDocs:
+        name: Website for the AsyncAPI Generator
+        url: https://www.asyncapi.com/tools/generator
+      requestBody:
+        description: Template details to be generated.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Template'
+      responses:
+        '200':
+          description: Template succesfully generated.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Template'
+        '422':
+          description: Failed to generate the given template due to invalid parameters or invalid AsyncAPI document.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        default:
+          description: Unexpected error.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+components:
+  schemas:
+    AsyncAPIDocument:
+      description: AsyncAPI document in JSON or YAML
+      type: string
+      minLength: 1
+    Template:
+      type: object
+      required:
+        - template
+      properties:
+        asyncapi:
+          $ref: '#/components/schemas/AsyncAPIDocument'
+        template:
+          type: string
+          description: Template name to be generated.
+          enum:
+            - 'html'
+            - 'markdown'
+        parameters:
```

Comment:
Shall we mention somehow that this maps directly with the parameters you will pass to the generator CLI `-p` arg? https://github.com/asyncapi/generator#cli-usage

Reply:
Description updated! :)